### PR TITLE
Fixing issues #7319 and #7244

### PR DESCRIPTION
**Pull request by kavishkafernando**
Fixes https://github.com/wso2/product-apim/issues/7319
Fixes https://github.com/wso2/product-apim/issues/7244

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 1]
```diff
@@ -5069,6 +5069,40 @@ public void deleteApplication(Application application) throws APIManagementExcep
...
             }
 
+            prepStmtGetConsumerKey = connection.prepareStatement(getConsumerKeyQuery);
+            prepStmtGetConsumerKey.setInt(1, application.getId());
+            rs = prepStmtGetConsumerKey.executeQuery();
+            List<String> consumerKeys = new ArrayList<>();
+
+            deleteDomainApp = connection.prepareStatement(deleteDomainAppQuery);
+            while (rs.next()) {
+                String consumerKey = rs.getString("CONSUMER_KEY");
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: You can use APIConstants.FIELD_CONSUMER_KEY

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 2]
```diff
@@ -5069,6 +5069,40 @@ public void deleteApplication(Application application) throws APIManagementExcep
...
+                    deleteDomainApp.addBatch();
+
+                    KeyManagerHolder.getKeyManagerInstance().deleteMappedApplication(consumerKey);
+                    // OAuth app is deleted if only it has been created from API Store. For mapped clients we don't
+                    // call delete.
+                    if (!"MAPPED".equals(mode)) {
+                        // Adding clients to be deleted.
+                        consumerKeys.add(consumerKey);
+                    }
+
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: extra line

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 3]
```diff
@@ -5069,6 +5069,40 @@ public void deleteApplication(Application application) throws APIManagementExcep
...
+                // This is true when OAuth app has been created by pasting consumer key/secret in the screen.
+                String mode = rs.getString("CREATE_MODE");
+                if (consumerKey != null) {
+                    deleteDomainApp.setString(1, consumerKey);
+                    deleteDomainApp.addBatch();
+
+                    KeyManagerHolder.getKeyManagerInstance().deleteMappedApplication(consumerKey);
+                    // OAuth app is deleted if only it has been created from API Store. For mapped clients we don't
+                    // call delete.
+                    if (!"MAPPED".equals(mode)) {
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: use OAuthAppMode.MAPPED

Comment (kavishkafernando) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java (hunk 2 above)
Comment (kavishkafernando) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java (hunk 1 above)
Comment (kavishkafernando) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Fixed

**Pull request merged**